src/Parsers/ASTPartition.h

```diff
@@ -15,6 +15,7 @@ class ASTPartition : public IAST
     size_t fields_count = 0;
 
     String id;
+    bool detach_all_partition = false;
```

Comment:
This field looks too specific. Can we use other condition to check if all were specified? E.g. just leave `value = nullptr` and `fields_count = 0` (but not sure that it's the best)

Reply:
I took your suggestion seriously. I refer to some existing code,ASTRolesOrUsersSet for example. A specific bool variable may be suitable. actually we can change it to "all", it is more comprehensible. fields_count is used to compare with fields count of partition key. value represents the child ASTPtr. Maybe using it confuses others.

```
class ASTRolesOrUsersSet : public IAST
{
public:
    bool all = false;
...
}
```

Reply:
but i have another problem. We execute "ALTER TABLE t DETACH PARTITION tuple()" when t has partition key, we could get a Exception:
`DB::Exception: Wrong number of fields in the partition expression: 0, must be: 1. (INVALID_PARTITION_VALUE)`,
maybe we could also skip the Exception.Because for no partition table ,it works.